Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs

```diff
@@ -75,25 +80,46 @@ public override void Initialize()
         }
 
-        private void AddFlushEjectVerbs(EntityUid uid, DisposalUnitComponent component, GetVerbsEvent<AlternativeVerb> args)
+        private void AddDisposalAltVerbs(EntityUid uid, DisposalUnitComponent component, GetVerbsEvent<AlternativeVerb> args)
         {
-            if (!args.CanAccess || !args.CanInteract || component.Container.ContainedEntities.Count == 0)
+            if (!args.CanAccess || !args.CanInteract)
                 return;
 
-            // Verbs to flush the unit
-            AlternativeVerb flushVerb = new();
-            flushVerb.Act = () => Engage(component);
-            flushVerb.Text = Loc.GetString("disposal-flush-verb-get-data-text");
-            flushVerb.IconTexture = "/Textures/Interface/VerbIcons/delete_transparent.svg.192dpi.png";
-            flushVerb.Priority = 1;
-            args.Verbs.Add(flushVerb);
-
-            // Verb to eject the contents
-            AlternativeVerb ejectVerb = new();
-            ejectVerb.Act = () => TryEjectContents(component);
-            ejectVerb.Category = VerbCategory.Eject;
-            ejectVerb.Text = Loc.GetString("disposal-eject-verb-contents");
-            args.Verbs.Add(ejectVerb);
+            // Behavior for if the disposals bin has items in it
+            if (component.Container.ContainedEntities.Count > 0)
+            {
+                // Verbs to flush the unit
+                AlternativeVerb flushVerb = new();
+                flushVerb.Act = () => Engage(component);
+                flushVerb.Text = Loc.GetString("disposal-flush-verb-get-data-text");
+                flushVerb.IconTexture = "/Textures/Interface/VerbIcons/delete_transparent.svg.192dpi.png";
+                flushVerb.Priority = 1;
+                args.Verbs.Add(flushVerb);
+
+                // Verb to eject the contents
+                AlternativeVerb ejectVerb = new();
+                ejectVerb.Act = () => TryEjectContents(component);
+                ejectVerb.Category = VerbCategory.Eject;
+                ejectVerb.Text = Loc.GetString("disposal-eject-verb-contents");
+                args.Verbs.Add(ejectVerb);
+            }
+
+            // Behavior if using a trash bag & other dumpable containers
+            if (args.Using != null
+                && EntityManager.HasComponent<DumpableComponent>(args.Using.Value)
+                && TryComp<DumpableComponent>(args.Using.Value, out var dumpable)
+                && TryComp<ServerStorageComponent>(args.Using.Value, out var storage)
+                && storage.StoredEntities != null
+                && storage.StoredEntities.Count > 0)
+            {
+                // Verb to dump held container into disposal unit
+                AlternativeVerb dumpVerb = new();
+                dumpVerb.Act = () => _entMan.EntitySysManager.GetEntitySystem<DumpableSystem>().StartDoAfter(args.Using.Value, args.Target, args.User, dumpable, storage);
```

Comment:
```suggestion
                dumpVerb.Act = () => _dumpableSystem.StartDoAfter(args.Using.Value, args.Target, args.User, dumpable, storage);
```
if you have the dependency there's no need to get the system through the entity manager.